mono/tests/test-runner.cs

```diff
@@ -186,8 +199,14 @@ class TestInfo {
 			string test = ti.test;
 			string opt_set = ti.opt_set;
 
-			if (concurrency == 1)
+			if (concurrency == 1) {
+				if (previous_test_hung) {
+					// The previous test hung and is being cleared up so give it a moment to finish
+					System.Threading.Thread.Sleep(1);
```

Comment:
The amount of waiting didn't seem to matter here in testing.  Without any waiting the new process will print out it's name before the old one prints it's exit status.